### DISCUSSION TITLE
How to copy engagements with REST API

### DISCUSSION THREAD

**Discussion by software-testing-professional** [DefectDojo/django-DefectDojo#7745] [Q&A]
Inside a DefectDojo product, we create separate engagements for each version of the product. 
So generally we have one engagement representing the current HEAD of the main branch. 
And a bunch of other engagements, representing release versions of this product: 

```
product
  + engagement "main"
  + engagement "v0.1.0"
  + engagement "v0.2.0"
```

Deduplication is done inside engagements only. 
Developers usually work on findings from the "main" engagement. 

If a new release is created (and therefore a new engagement), we would like to create this engagement by copying the "main" engagement. Including all the findings with risk acceptance etc. that have already been processed in the "main" engagement. 

The DefectDojo UI provides a "Copy" function for engagements. 
But I can't find the matching request in the Swagger API description. 

How can "copy engagement" be done via REST API?

Best regards, Michael

**Comment by Hertog-PJ** [comment]
Did you find any solution for this?

We want to go the same route with how we use engagements, but it would be highly preferable if we can use an API to do the copy step to facilitate automation.

**Comment by valentijnscholten** [comment]
Currently this is not possible @software-testing-professional @Hertog-PJ 
Please raise a feature request via https://github.com/DefectDojo/django-DefectDojo/issues/new?template=feature_request.md
